src/app/components/planner/plannerTabs/PlannerTabs.tsx

```diff
@@ -0,0 +1,230 @@
+      alertOpen: true,
+    });
+  };
+
+  const handleBlur = (title: string) => {
+    title.length >= 2 && setPlannerBeingEdited(null);
+  };
+
+  const handleTabChange = (index: number | string | null) => {
+    if (typeof index === "string" && index !== "add-planner-tab") {
+      const title = planners[index][0];
+      switchPlanners(index, title);
+    }
+  };
+
+  return (
+    <>
+      <div className="grid grid-flow-col gap-2 ml-1 overflow-x-auto">
+        {Object.entries(planners).map(([id, [title]]) => (
+          <CustomTab
+            key={id}
+            title={title}
+            id={id}
+            selected={activePlanner ? activePlanner.id === id : false}
+            isEditing={plannerBeingEdited === id}
+            setPlannerBeingEdited={setPlannerBeingEdited}
+            onEndEditing={(newTitle) => {
+              changePlannerName(newTitle, id);
+              handleBlur(newTitle);
+            }}
+            onClick={() => handleTabChange(id)}
+            onOpenDeleteAlert={handleOpenDeleteAlert}
+          />
+        ))}
+        <Button
+          loading={plannersLoading}
+          aria-label="Add"
+          onClick={() => handleAddPlanner()}
+          size="sm"
+          variant="plain"
+          color="primary"
+          startDecorator={<Add />}
+        />
+      </div>
+      <ConfirmAlert
+        open={deleteAlert.alertOpen}
+        onClose={() => setDeleteAlert(emptyDeleteAlertData)}
+        onConfirm={() => {
+          deletePlanner(deleteAlert.id);
+          setDeleteAlert(emptyDeleteAlertData);
+        }}
+        dialogText={deleteAlert.title}
+      />
+      <TooManyPlannersAlert
+        open={tooManyAlertIsOpen}
+        onClose={() => setTooManyAlertIsOpen(false)}
+      />
+    </>
+  );
+}
+
+function CustomTab({
+  title,
+  id,
+  isEditing,
+  selected,
+  setPlannerBeingEdited,
+  onEndEditing,
+  onClick,
+  onOpenDeleteAlert,
+}: {
+  title: string;
+  id: string;
+  isEditing: boolean;
+  selected: boolean;
+  setPlannerBeingEdited: (id: string) => void;
+  onEndEditing: (newTitle: string) => void;
+  onClick: () => void;
+  onOpenDeleteAlert: (id: string, title: string) => void;
+}) {
+  const [text, setText] = useState(title);
+  const [hovering, setHovering] = useState(false);
+
+  function backgroundColor() {
+    if (hovering) {
+      return "#BFDBFE";
+    } else if (selected) {
+      return "#3B82F6";
+    } else {
+      return "#93C5FD";
+    }
+  }
+
+  function textColor() {
+    if (hovering) {
+      return "#334155";
+    } else if (selected) {
+      return "#F1F5F9";
+    } else {
+      return "#334155";
+    }
+  }
+
+  function handleClick() {
+    if (selected) {
+      setPlannerBeingEdited(id);
+    } else {
+      onClick();
+    }
+  }
+
+  const cursorStyle = hovering ? { cursor: "pointer" } : {};
+
+  return (
+    <div
+      className="p-2 rounded-xl flex items-center justify-between"
```

Comment:
Maybe try making the tab rounded-full. The tab looks kind of blocky as it is right now.